pyvista/core/utilities/fileio.py

```diff
@@ -552,27 +565,34 @@ def save_meshio(filename, mesh, file_format=None, **kwargs):
     # Get cells
     cells = []
     c = 0
-    for offset, cell_type in zip(vtk_offset, vtk_cell_type):
-        numnodes = vtk_cells[offset + c]
-        cell = vtk_cells[offset + 1 + c : offset + 1 + c + numnodes]
-        c += 1
-        cell = (
-            cell
-            if cell_type not in pixel_voxel
-            else cell[[0, 1, 3, 2]]
-            if cell_type == 8
-            else cell[[0, 1, 3, 2, 4, 5, 7, 6]]
-        )
-        cell_type = cell_type if cell_type not in pixel_voxel else cell_type + 1
-        cell_type = vtk_to_meshio_type[cell_type] if cell_type != 7 else f"polygon{numnodes}"
+    for i, (offset, cell_type) in enumerate(zip(vtk_offset, vtk_cell_type)):
+        if cell_type == 42:
+            cell_ = mesh.get_cell(i)
```

Comment:
Not sure if there is a more efficient way to do that without having to deal with a deep copy. I don't see any grid attribute that exposes the faces of a cell.